miden-lib/src/notes/mod.rs

```diff
@@ -38,8 +38,10 @@ pub fn create_p2id_note<R: FeltRng>(
     let inputs = [target.into()];
     let tag: Felt = target.into();
     let serial_num = rng.draw_word();
+    let aux = Felt::ZERO;
```

Comment:
We should probably change how we set the tag here - otherwise, it could cause issues?

Reply:
what do you mean? do you want this to be an argument?

Reply:
In https://github.com/0xPolygonMiden/miden-base/issues/402#issuecomment-2008397166 I described how the tags could be set. The main change here is that instead of setting the tag to the full account ID (the `target`) we should set it to the appropriate tag prefix (e.g., `00`) + the first 16 bit the target + padded with zeros to fill the remaining bits (up to 32).

Reply:
and how do you want this to look? this will affect all the bit manipulations. should the 32 bits be in the most or least significant part of the word?

Reply:
ah, I see, the comment was on the line for the `aux`, but you meant the `tag`. Yeah, I will have to fix that in a follow up